### PR TITLE
Add new topologies

### PR DESCRIPTION
**Pull request by arnabGudu**

### CODE REVIEW

File: api/app.py [hunk 1]
```diff
@@ -28,24 +28,68 @@
...
+            status_code=status.HTTP_400_BAD_REQUEST, 
+            detail="product_id and price fields are required")
+    document = {
+        "product_id": product_id, "price": price}
     result = collection.insert_one(document)
-    return str(result.inserted_id)
 
 
 kafka_map = {
+    'updateAttribute': {'fields': ['attribute', 'handler': None]},
```
Comment (KSSSenapati) [inline api/app.py:84]: `attribute` to `attributes`

File: api/app.py [hunk 2]
```diff
@@ -28,24 +28,68 @@
...
-    'updateSKU': {'fields': ['sizesCount'], 'handler': None},
+    'updateSKU': {'fields': ['sizesCount'], 'handler': sku_handler},
     'updateRating': {'fields': ['rating'], 'handler': None},
-    'updateTag': {'fields': ['accelerator_tag'], 'handler': None},
+    'updateTag': {'fields': ['accelerator_tag'], 'handler': tag_handler},
     'updateAd': {'fields': ['ad_enabled'], 'handler': None},
     'updateImage': {'fields': ['main_image'], 'handler': image_handler},
     'updateProduct': {'fields': ['product_title', 'master_category', 'sub_category',
-        'article', 'attribute', 'price', 'pincode'], 'handler': None},
+        'article', 'price', 'pincode'], 'handler': price_handler},
```
Comment (KSSSenapati) [inline api/app.py:92]: `article` to `article_type`

File: api/app.py [hunk 3]
```diff
@@ -28,24 +28,68 @@
...
+            detail="product_id and price fields are required")
+    document = {
+        "product_id": product_id, "price": price}
     result = collection.insert_one(document)
-    return str(result.inserted_id)
 
 
 kafka_map = {
+    'updateAttribute': {'fields': ['attribute', 'handler': None]},
     'updateDiscount': {'fields': ['discount', 'sale_discount'], 'handler': None},
```
Comment (KSSSenapati) [inline api/app.py:85]: please include `discounted_price` & `sale_discounted_price` as well

File: api/app.py (hunk 2 above)
Comment (KSSSenapati) [reply api/app.py:92]: remove hardcoded product_id from `/addProduct`
make it. to `body.get("product_id")`

File: storm_topology/src/bolts/DiscountBolt.py [hunk 4]
```diff
@@ -1,16 +1,18 @@
...
         discount = message.get('discount')
-        if discount:
+        sale_discount = message.get('sale_discount')
+        if all(val is not None for val in [product_id, price, discount, sale_discount]):
             discounted_price = price * (1 - (discount / 100))
-            self.emit({'product_id': product_id, 'price': discounted_price})
+            sale_discounted_price = price * (1 - (sale_discount / 100))
+            update = {'discount': 'set', 'discounted_price': 'set', 'sale_discounted_price': 'set'}
+            doc = {'id': f'doc_{product_id}', 'discount': discount, 
+                   'discounted_price': discounted_price, 'sale_discounted_price': sale_discounted_price}
```
Comment (KSSSenapati) [inline storm_topology/src/bolts/DiscountBolt.py:17]: add `sale_discount` as well to both doc and update dictionary

File: storm_topology/src/bolts/SKUBolt.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+class SKUBolt(Bolt):
+    def initialize(self, stormconf, context):
+        pass
+
+    def process(self, tup):
+        product_id, sku = tup.values[0]
+        if product_id is not None and sku is not None:
+            doc = {'id': f'doc_{product_id}'}
+            update['sizesCount'] = 'set'
```
Comment (KSSSenapati) [inline storm_topology/src/bolts/SKUBolt.py:11]: change `update['sizesCount'] = 'set'` to `update['sizes_facet'] = 'set'`

**Review by KSSSenapati — CHANGES_REQUESTED** [review]
Please resolve comments

**Comment by arnabGudu** [comment]
Resolved all comments

**Pull request merged**